src/ssh/cockpitsshrelay.c

```diff
@@ -938,7 +942,15 @@ do_key_identity_auth (CockpitSshData *data)
       if (rc != SSH_AUTH_SUCCESS)
         {
           g_debug ("%s isn't accepted by the server", identity);
-          goto out;
+        try_next_identity:
+          if (default_identities[next_identity] == NULL)
+            goto out;
+          ssh_key_free (pub_key);
+          pub_key = NULL;
+          ssh_string_free_char (identity);
+          g_free (pub_key_path);
+          identity = g_strdup_printf ("%s/.ssh/%s", g_get_home_dir (), default_identities[next_identity++]);
```

Comment:
what happens here if you try an identity which doesn't exist? I guess it will just fail and go to the next one?

Reply:
Yes, exactly.